frontend/packages/dev-console/src/components/pipelineruns/PipelineTaskLogs.tsx

```diff
@@ -0,0 +1,168 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { K8sKind } from '@console/internal/module/k8s';
+import {
+  ResourceLog,
+  LOG_SOURCE_TERMINATED,
+  LoadingInline,
+} from '@console/internal/components/utils';
+import { ContainerStatus, containerToLogSourceStatus } from '../../utils/pipeline-utils';
+import './PipelineTaskLogs.scss';
+
+interface Pod {
+  data: K8sKind;
+}
+
+interface Container extends K8sKind {
+  name?: string;
+}
+
+interface PipelineTaskLogsProps {
+  obj?: Pod;
+  taskName: string;
+}
+
+interface PipelineTaskLogsState {
+  renderContainers: Container[];
+  fetching: boolean;
+  targetHeight: number;
+}
+
+class PipelineTaskLogs extends React.Component<PipelineTaskLogsProps, PipelineTaskLogsState> {
+  scrollPane: React.RefObject<HTMLDivElement> = React.createRef();
+
+  scrollUntouched: boolean = true;
+
+  isAutoScrolling: boolean = false;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      renderContainers: [],
+      fetching: false,
+      targetHeight: null,
+    };
+  }
+
+  componentDidMount() {
+    const containers: Container[] = _.get(this.props, ['obj', 'data', 'spec', 'containers'], []);
+    const containerStatus: ContainerStatus[] = _.get(
+      this.props,
+      ['obj', 'data', 'status', 'containerStatuses'],
+      [],
+    );
+    const firstrunningCont = containerStatus.findIndex(
+      (container) => containerToLogSourceStatus(container) !== LOG_SOURCE_TERMINATED,
+    );
+    const renderContainers = containers.slice(
+      0,
+      firstrunningCont === -1 ? containers.length : firstrunningCont + 1,
+    );
+
+    this.setState({ renderContainers });
+    firstrunningCont === -1
+      ? this.setState({ fetching: false })
+      : this.setState({ fetching: true });
+    this.scrollToBottom();
+    window.addEventListener('resize', this.sizeContainer, { passive: true });
+    this.sizeContainer();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const containers: Container[] = _.get(nextProps, ['obj', 'data', 'spec', 'containers'], []);
+    const containerStatus: ContainerStatus[] = _.get(
+      nextProps,
+      ['obj', 'data', 'status', 'containerStatuses'],
+      [],
+    );
+    if (this.props !== nextProps) {
```

Comment:
I don't think we can compare props directly. It is going to be true always.

Reply:
@abhinandan13jan use can `isEqual` instead.
```suggestion
    if (_.isEqual(this.props, nextProps)) {
```